src/recipes/login.tsx

```diff
@@ -0,0 +1,294 @@
+import { StyleSheet, View } from "react-native";
+import * as React from "react";
+import { Button, Controls, Input, Select } from "../components";
+import { colors } from "../theme";
+import Touchable from "../components/shared/Touchable";
+import Countdown from "../components/shared/Countdown";
+import Text from "../components/Text";
+import OTPInput from "react-native-otp";
+
+interface OperationalCountry {
+  id: number;
+  name: string;
+  url_name: string;
+  country_code: string;
+}
+
+interface LoginProps {
+  loginUserValue: string;
+  onLoginUserChange: (value: string) => void;
+  onSendOtp: (
+    val: "phone" | "email",
+    onSuccess: () => void,
+    onError: () => void
+  ) => void;
+  otpValue: string;
+  onOtpChange: (value: string) => void;
+  onResendOtp: (val: "phone" | "email") => void;
+  onSignIn: () => void;
+  countriesList: OperationalCountry[];
+  onCountryChange: (country: OperationalCountry) => void;
+  selectedCountry: number;
+  footer?: React.ReactText | JSX.Element;
+  otpLength: number;
+}
+
+interface LoginState {
+  loginMethod: number;
+  loginPage: number;
+  sendingOTP: boolean;
+  otpTimeout: boolean;
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    justifyContent: "center",
+    padding: 25
+  },
+  loginSubHeader: {
+    marginTop: 15
+  },
+  formContainer: {
+    marginTop: 30
+  },
+  loginUserInput: {
+    marginTop: 60,
+    flexDirection: "row"
+  },
+  userInfoWrap: {
+    flexDirection: "row",
+    alignItems: "center"
+  },
+  loginUserText: { flexGrow: 1 },
+  textButton: {
+    padding: 10,
+    color: colors.violet.base,
+    fontWeight: "bold"
+  },
+  otpInput: { marginRight: 20, flexShrink: 1, marginBottom: 0 },
+  countrySelect: { width: 100, marginRight: 30 },
+  phoneInput: { flex: 1 },
+  otpInputWrap: {
+    flexDirection: "row",
+    alignItems: "center",
+    justifyContent: "space-between"
+  },
+  countdownStyles: { marginTop: 10, fontWeight: "bold", marginHorizontal: 10 },
+  cellStyle: {
+    borderWidth: 0,
+    borderBottomWidth: 1,
+    margin: 0,
+    marginRight: 20,
+    paddingVertical: 0,
+    paddingBottom: 5
+  }
+});
+
+enum LOGIN_OPTIONS {
+  PHONE = 1,
+  EMAIL
+}
+
+enum LOGIN_PAGE {
+  USER_PAGE = 1,
+  OTP_PAGE
+}
+
+const LOGIN_METHODS = [
+  { id: LOGIN_OPTIONS.PHONE, name: "Phone Number" },
+  { id: LOGIN_OPTIONS.EMAIL, name: "Email Address" }
+];
+
+export default class Login extends React.PureComponent<LoginProps, LoginState> {
+  state = {
+    loginMethod: LOGIN_OPTIONS.PHONE,
+    loginPage: LOGIN_PAGE.USER_PAGE,
+    sendingOTP: false,
+    otpTimeout: false
+  };
+
+  onOtpSuccess = () =>
+    this.setState({
+      sendingOTP: false,
+      loginPage: LOGIN_PAGE.OTP_PAGE
+    });
+
+  onOtpError = () => this.setState({ sendingOTP: false });
```

Comment:
also, call something like `this.props.onOtpError` in case the user wants to show a toast.